modules/schema/src/main/java/org/apache/ignite/internal/schema/BinaryTupleComparatorUtils.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.schema;
+
+import static java.lang.Integer.signum;
+import static org.apache.ignite.internal.binarytuple.BinaryTupleCommon.EQUALITY_FLAG;
+import static org.apache.ignite.internal.lang.IgniteStringFormatter.format;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import org.apache.ignite.internal.binarytuple.BinaryTupleReader;
+import org.apache.ignite.internal.type.NativeTypeSpec;
+
+/**
+ * The utility class has methods to use to compare fields in binary representation.
+ */
+class BinaryTupleComparatorUtils {
+    /**
+     * Compares individual fields of two tuples using ascending order.
+     */
+    @SuppressWarnings("DataFlowIssue")
+    static int compareFieldValue(NativeTypeSpec typeSpec, BinaryTupleReader tuple1, int index1, BinaryTupleReader tuple2, int index2) {
```

Comment:
This method is always called with index1 == index2.
You can remove index2 argument.

Reply:
It is a copy of the code that had been written before.
We can use only one column index for both tuples.

Reply:
Fixed.